### DISCUSSION TITLE
Docker Build Fails with Timeout Despite Having Local Image - How to Force Docker to Use Local Image Instead of Pulling from Docker Hub?

### DISCUSSION THREAD

**Discussion by OVELYB** [kubernetes-sigs/scheduler-plugins#833] [Q&A]
I want to build the images of the scheduler-plugins with my example plugin.  I use the **Makefile** in the project.
My go version is 1.23.2 and I have already downloaded the required base image golang:1.23.2 locally. However, when I run the build command--_**make build-images**_, Docker still attempts to pull the image from Docker Hub and fails due to network timeout errors.

The error I receive is:

ERROR: failed to solve: golang:1.23.2: failed to resolve source metadata for docker.io/library/golang:1.23.2: failed to do request: Head "https://registry-1.docker.io/v2/library/golang/manifests/1.23.2": dial tcp 174.37.175.229:443: i/o timeout

I have verified that the local golang:1.23.2 image exists on my system using docker images, and I can see the image listed. However, Docker seems to ignore the local image and is trying to pull it from the remote Docker Hub repository.

Here are the steps I’ve already taken:

I’ve tried specifying the local image in the Dockerfile and in the build command, but it still pulls from the remote registry.I tried adding the --no-cache option to the docker buildx build command to force a fresh build, but the issue persists. I also checked my Docker configuration to make sure it's set to prefer local images, but it doesn’t seem to work.

**My problem is: What could be causing Docker to still pull the image from the remote registry despite having it locally, and how can I ensure that Docker uses the local image instead?**

**Comment by Huang-Wei** [comment]
I'm wondering is it possible to use some HTTP proxy to work this around?

**Reply by prb112** [reply]
Hey @OVELYB go ahead and ping me ... based on reverse dns, I think you are doing this on SoftLayer ... You can drop me a message at pbastide@us.ibm.com and we can share a screen - I would think your setup needs some tweaking. Thanks, Paul
